internal/lsp/server.go

```diff
@@ -2132,7 +2119,7 @@ func (l *LanguageServer) loadWorkspaceContents(ctx context.Context, newOnly bool
 		}
 
 		// if the caller has requested only new files, then we can exit early
-		if _, ok := l.cache.GetModule(fileURI); newOnly && ok {
+		if _, ok := l.cache.GetFileContents(fileURI); newOnly && ok {
```

Comment:
The best proxy for 'Existence in the cache' is file contents and not a parsed module. This is related to the fix, but is in addition to the required change.